sormas-backend/src/main/java/de/symeda/sormas/backend/caze/CaseFacadeEjb.java

```diff
@@ -2094,24 +2092,60 @@ private void updatePersonAndCaseByOutcome(CaseDataDto existingCase, Case newCase
 				}
-			} else if (newCase.getOutcome() == CaseOutcome.NO_OUTCOME) {
-				if (newCase.getPerson().getPresentCondition() != PresentCondition.ALIVE) {
-					PersonDto existingPerson = PersonFacadeEjb.toDto(newCase.getPerson());
-					newCase.getPerson().setPresentCondition(PresentCondition.ALIVE);
-					// attention: this may lead to infinite recursion when not properly implemented
-					personFacade.onPersonChanged(existingPerson, newCase.getPerson());
+
+			} else if ((newCase.getOutcome() == CaseOutcome.NO_OUTCOME || newCase.getOutcome() == CaseOutcome.RECOVERED)
+				&& existingCase.getOutcome() == CaseOutcome.DECEASED) {
+				// Case was put "back alive"
+				PersonDto existingPerson = PersonFacadeEjb.toDto(newCase.getPerson());
+				boolean dateThreshold = newCase.getReportDate().before(DateHelper.addDays(existingPerson.getDeathDate(), 30))
+					&& newCase.getReportDate().after(DateHelper.subtractDays(existingPerson.getDeathDate(), 30));
+
+				if (existingPerson.getCauseOfDeath() == CauseOfDeath.EPIDEMIC_DISEASE
+					&& existingPerson.getCauseOfDeathDisease() == newCase.getDisease()
+					&& dateThreshold) {
+					// Make sure no other case associated with the person has Outcome=DECEASED
+					if (getAllCasesOfPerson(existingPerson.getUuid()).stream()
+						.noneMatch(caze -> caze.getOutcome() == CaseOutcome.DECEASED && !caze.getUuid().equals(existingCase.getUuid()))) {
+						newCase.getPerson().setPresentCondition(PresentCondition.ALIVE);
+						newCase.getPerson().setBurialDate(null);
+						newCase.getPerson().setDeathDate(null);
+						newCase.getPerson().setDeathPlaceDescription(null);
+						newCase.getPerson().setDeathPlaceType(null);
+						newCase.getPerson().setCauseOfDeath(null);
+						newCase.getPerson().setCauseOfDeathDetails(null);
+						newCase.getPerson().setCauseOfDeathDisease(null);
+						personFacade.onPersonChanged(existingPerson, newCase.getPerson());
+					}
 				}
 			}
 		} else if (existingCase != null
-			&& CaseOutcome.DECEASED == newCase.getOutcome()
+			&& newCase.getOutcome() == CaseOutcome.DECEASED
 			&& (newCase.getPerson().getPresentCondition() == PresentCondition.DEAD
 				|| newCase.getPerson().getPresentCondition() == PresentCondition.BURIED)
-			&& (newCase.getPerson().getDeathDate() == null
-				? newCase.getOutcomeDate() != null
-				: !newCase.getPerson().getDeathDate().equals(newCase.getOutcomeDate()))) {
-			PersonDto existingPerson = PersonFacadeEjb.toDto(newCase.getPerson());
-			newCase.getPerson().setDeathDate(newCase.getOutcomeDate());
-			personFacade.onPersonChanged(existingPerson, newCase.getPerson());
+			&& existingCase.getOutcomeDate() != newCase.getOutcomeDate()
+			&& newCase.getOutcomeDate() != null) {
+			// outcomeDate of a deceased case was changed, but person is already considered dead
+			if (newCase.getOutcomeDate() != null
+				&& newCase.getPerson().getCauseOfDeath() == CauseOfDeath.EPIDEMIC_DISEASE
```

Comment:
Move these checks to the if clause above to prevent unnecessary nested if clauses